app/controllers/contacts_controller.rb

```diff
@@ -0,0 +1,21 @@
+class ContactsController < ApplicationController
+  def new  
+    @company = Company.find(params[:id])
+  end
+
+  def create
+    @contact = Contact.new(contact_params)
+    @contact.company_id = params[:company_id]
```

Comment:
leverage your associations here rather than setting the foreign key manually. you should be able to do the following:

```ruby
@company = Company.find(params[:company_id])
@contact = @company.contacts.new(contact_params)
```